src/parsers/nullable.ts

```diff
@@ -47,6 +47,9 @@ export function parseNullableDef(
       currentPath: [...refs.currentPath],
     });
 
+    if (base && '$ref' in base)
+        return { allOf: [base], nullable: true } as any;
```

Comment:
Moving the result into allOf will affect its seen path, so need to account for this to not break references to this node

Reply:
It should be safe to always use the allOf stuff so maybe we can assume that's where it will end up

Reply:
> Moving the result into allOf will affect its seen path, so need to account for this to not break references to this node

I am not too experienced with json schemas. Is it allowed to reference a reference? And will it ever happend?

Reply:
> In Draft 4-7, $ref behaves a little differently. When an object contains a $ref property, the object is considered a reference, not a schema. Therefore, any other properties you put in that object will not be treated as JSON Schema keywords and will be ignored by the validator. $ref can only be used where a schema is expected.
[https://json-schema.org/understanding-json-schema/structuring#dollarref](https://json-schema.org/understanding-json-schema/structuring#dollarref)

"$ref can only be used where a schema is expected."